packages/components/src/components/divider/divider.component.ts

```diff
@@ -0,0 +1,262 @@
+    if (!buttonElement) return;
+
+    const iconType = this.getArrowIcon();
+    buttonElement.setAttribute('variant', 'secondary');
+    buttonElement.setAttribute('aria-label', this.ariaLabel || '');
+    buttonElement.setAttribute('aria-expanded', this.ariaExpanded || '');
+    buttonElement.setAttribute('prefix-icon', iconType);
+  }
+
+  /**
+   * Sets the `aria-label` attribute for label based on `aria-label` of the text divider.
+   * @returns {void}
+   */
+  private setTextDivider(): void {
+    const textElement = this.querySelector('mdc-text');
+    if (!textElement) return;
+
+    textElement.setAttribute('aria-label', this.ariaLabel || '');
+  }
+
+  /**
+   * Determines the arrow icon based on the consumer-defined `arrowDirection`.
+   *
+   * @returns {string} The icon that represents the arrow direction.
+   */
+  private getArrowIcon(): string {
+    // set fallback class
+    let arrowIcon = this.orientation === DIVIDER_ORIENTATION.HORIZONTAL ? 'arrow-down-regular' : 'arrow-right-regular';
+
+    if (this.orientation === DIVIDER_ORIENTATION.HORIZONTAL) {
+      arrowIcon = this.arrowDirection === DIRECTIONS.POSITIVE ? 'arrow-up-regular' : 'arrow-down-regular';
+    } else {
+      arrowIcon = this.arrowDirection === DIRECTIONS.POSITIVE ? 'arrow-right-regular' : 'arrow-left-regular';
+    }
+    return arrowIcon;
+  }
+
+  public override update(changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
+    super.update(changedProperties);
+
+    if (changedProperties.has('dividerTypeInternal')) {
+      this.setClassBasedOnType(this.dividerTypeInternal);
+    }
+
+    if (changedProperties.has('orientation')) {
+      this.setOrientation(this.orientation);
+    }
+
+    if (changedProperties.has('variant')) {
+      this.setVariant(this.variant);
+    }
+
+    if (this.dividerTypeInternal === DIVIDER_TYPE_INTERNAL.TEXT && changedProperties.has('ariaLabel')) {
+      this.setTextDivider();
+    }
+
+    if (this.dividerTypeInternal === DIVIDER_TYPE_INTERNAL.GRABBER_BUTTON) {
+      if (
+        changedProperties.has('orientation')
+        || changedProperties.has('arrowDirection')
+        || changedProperties.has('buttonPosition')
+        || changedProperties.has('ariaLabel')
+        || changedProperties.has('ariaExpanded')
+      ) {
+        this.setGrabberButton();
+      }
+    }
+  }
+
+  /**
+   * Sets the class of the Divider based on type.
+   * @param type - The type of the divider.
+   */
+  private setClassBasedOnType(type: string) {
+    this.classList.remove('mdc-primary-divider', 'mdc-text-divider', 'mdc-grabber-divider');
+    if (type === DIVIDER_TYPE_INTERNAL.PRIMARY) {
+      this.classList.add('mdc-primary-divider');
+    } else if (type === DIVIDER_TYPE_INTERNAL.TEXT) {
+      this.classList.add('mdc-text-divider');
+    } else if (type === DIVIDER_TYPE_INTERNAL.GRABBER_BUTTON) {
+      this.classList.add('mdc-grabber-divider');
+    }
+  }
```

Comment:
issue: since we are setting class names on the component, it is treated similar to having classes on an HTML element. The consumers have an option to remove these classes. Thus, we shouldn't depend on class names for our component logic.
suggestion: We can make use of data attributes. Setting data attributes on the element itself can be helpful. We can further use this attributes for styling.

```suggestion
  private setClassBasedOnType(type: string) {
    if (type === DIVIDER_TYPE_INTERNAL.PRIMARY) {
      this.setAttribute('data-type','mdc-primary-divider');
    } else if (type === DIVIDER_TYPE_INTERNAL.TEXT) {
        this.setAttribute('data-type','mdc-text-divider');
    } else if (type === DIVIDER_TYPE_INTERNAL.GRABBER_BUTTON) {
      this.setAttribute('data-type','mdc-grabber-divider');
    }
  }
```

On the styles file, you can use them just like any other attribute
```
:host([data-type='mdc-text-divider']){
// styles
}
```